### ISSUE TITLE
Tweak Published UI of My Workflows/Tools

### ISSUE THREAD

**Issue by coverbeck** [ga4gh/dockstore#1094]
## Feature Request 

On the My Tools/Workflows page, there is a panel on the top left that lists all your tools or workflows. The panel has blue labels with the word `PUB` to indicate published, and `UNPUB` to indicate unpublished.

When I first came to the page, I thought I could click on them to Publish or Unpublish, because the labels look very much like buttons. But the labels indicate the state you are in, not the state to go to.

### Desired behaviour

I suggest:

1. Change the style to be a badge
2. For published tools/workflows, change the color to be green instead of blue, as green often indicates success.
3. Optionally, not have a badge at all for the Unpublished items. The drawback to this approach is that you haven't published anything, then nothing will be visible, and it may not be apparent that you need to publish. On the other hand, if you think of it from a badge perspective, then you just haven't added the published badge yet.
4. If we do 3, then spell out `Published`. If we don't do 3, then we may have to stick with `Pub`, because there might not be enough room to also have `Unpublished`.

See screenshot, where the last two items have badges instead of labels/buttons. In this screenshot, I do have a badge for Unpub, although I think we should remove it.

<img width="272" alt="screen shot 2018-01-08 at 3 41 08 pm" src="https://user-images.githubusercontent.com/1049340/34698196-b8129b44-f48b-11e7-952b-98aac3bac266.png">

**Comment by denis-yuen** [comment]
1) This probably draws inspiration from somewhere specific, but rounding the label doesn't really say "badge" to me. Reading from the top, I though you meant something like the style at https://shields.io/ 
2) This might be something to discuss as a team. At first blush, I'm not sure a published/not-published tool fits nicely into success/failure. Having a red failure badge would seem to encourage a fix where none may potentially be needed. The no badge idea in 3 might work. 

4) Spelling out published might work.

**Comment by coverbeck** [comment]
1. I was referencing Bootstrap 3 [badges](https://getbootstrap.com/docs/3.3/components/#badges), and used that badge class to create the screenshot. Their examples use numbers.  I've think I've used it in the past for text, but maybe this is not a good way to use it after all.
2. Agreed that not-published is not necessarily a failure, which argues that we should not have a badge for not published.

My concern was that as a new user I thought at first view that Pub/Unpub were clickable buttons.

**Comment by denis-yuen** [comment]
Oh, I see. In the linked bootstrap examples, its a bit more obvious when using it as an unread email count or a notification count.

**Comment by coverbeck** [comment]
I've attached a screenshot with 3 options, from top to bottom:

1. The way it is today
2. Removing UNPUB indicator and changing PUB indicator color to green
3. Removing UNPUB indicator, changing PUB to PUBLISHED, and changing color to green.

I got rid of the Bootstrap badge styling I had in my previous screenshot.

Is 2 or 3 any more appealing, or do you feel we should just leave it as is?

<img width="817" alt="screen shot 2018-01-09 at 1 36 01 pm" src="https://user-images.githubusercontent.com/1049340/34745181-75a371f2-f544-11e7-8b5e-362dcadc97b6.png">

**Comment by denis-yuen** [comment]
I'm thinking "Removing UNPUB indicator, changing PUB to PUBLISHED, but leaving the colour." For some reason, the green colour looks a bit jarring to me (especially if there were a lot of them), but it might be best to bring this to the group to get a broader opinion

@garyluu  @agduncan94  ?

**Comment by coverbeck** [comment]
@denis-yuen Here is a screen shot with your proposal.

<img width="271" alt="screen shot 2018-01-09 at 3 11 21 pm" src="https://user-images.githubusercontent.com/1049340/34747990-6ca85ddc-f54f-11e7-96d9-98c6c8532d44.png">

**Comment by agduncan94** [comment]
**1. Colour of the badge/tag**
I do think it is a bit weird to use the blue colour for both the published badge and the +/refresh buttons in the bottom row of the org box. I agree with the idea that green might not be the best colour for this, as it would imply an unpublished entry is something negative. Perhaps we need to come up with some standards to represent these states, and not just use the bootstrap defaults.

**2. Content of the badge/tag**
The problem with having the full word for published cuts out some space for the entry repository. This can be a problem if you have multiple entries that have the same repository but have different toolnames/workflownames. This can be seen in the above comment by coverbeck. There are multiple dockstore-workflow-md5s... but you can't differentiate them by looking at the side bar. Perhaps a quick fix for this would be to have a tooltip showing the full name of the entry.

**3. Removal of unpublished tag/badge**
I do agree that the unpublished tag/badge is not necessary.


In summary, I think we should drop the unpub, and find some sort of icon/colour to represent the published state of an entry. I think the Published tag/badge just takes up too much space, though a tooltip entries with longer names might be a temporary way to deal.

**Comment by coverbeck** [comment]
I still like green, as I feel it's a common indicator for success, and the lack of green does not indicate failure -- if anything, I think the current `Unpub` which is orange/yellow, implies a failure more than my proposal does.

However, since no one's in agreement with me, how about for now we just drop `Unpub` and leave everything else the same? I think we are in agreement on that one.

**Comment by agduncan94** [comment]
I think I'd prefer the green over the current colour (blue), but seems like @denis-yuen prefers the blue.  @garyluu thoughts?

Github really needs some sort of poll feature.

**Comment by denis-yuen** [comment]
Agreed on dropping `Unpub`, I still think the green is a bit too eye catching, like the `Comment` button below.

**Comment by keiranmraine** [comment]
Sorry to add to this, but could the published/unpublished be an expandable list rather than a name sorted one with tags at the end?

I have quite a few un-published things that persist as I can't delete them.  Would be good if they were just hidden in a sub tree.  Currently I have this:

![screen shot 2018-02-08 at 09 30 11](https://user-images.githubusercontent.com/3740323/35965428-bb9db430-0cb2-11e8-944c-00168aa797b1.png)

It would be good if we could have something like the following, it would potentially reduce use of horizontal space having it as a heading rather than a "badge" on each repo:

```
.
└── quay.io
    └── wtsicgp
        ├── published
        │   ├── dockstore-biobambam2/bamtofastq
        │   ├── dockstore-cgp-chksum
        │   ├── dockstore-cgp-seqval
        │   ├── dockstore-cgpmap
        │   ├── dockstore-cgpmap/cgpmap-bamOut
        │   ├── dockstore-cgpmap/cgpmap-cramOut
        │   ├── dockstore-cgpwgs
        │   └── dockstore-cgpwxs
        └── unpub
            ├── cgp-somatic-docker
            ├── cgp_in_a_box
            ├── cgpbox
            ├── docker-cgp-ngs-qc
            ├── dockstore-biobambam2/biobambam2-bamtofastq
            ├── dockstore-cgp-irap
            └── dockstore-cgpmap/bamOutput
```

**Comment by coverbeck** [comment]
I like this idea. On top of that, I also find it confusing and noisy to see repos that don't even have workflows in the list, with an Unpub badge next to them, implying that I could publish them. Following your idea, maybe `unpub` could be collapsed by default.

**Comment by denis-yuen** [comment]
Makes sense

**Comment by denis-yuen** [comment]
Working as a selectable tab separating published and unpublished tools/workflows
